File: HTTPShortcuts/app/src/main/java/ch/rmy/android/http_shortcuts/utils/Validation.java
package ch.rmy.android.http_shortcuts.utils;

import android.text.TextUtils;

public class Validation {

    public static boolean isEmpty(String string) {
        return string.matches("^\\s*$");
    }

    public static boolean isAcceptableUrl(String url) {
        return !TextUtils.isEmpty(url) && !url.equalsIgnoreCase("http://") && !url.equalsIgnoreCase("https://");
    }

}

File: HTTPShortcuts/app/src/main/java/ch/rmy/android/http_shortcuts/utils/Validation.kt
package ch.rmy.android.http_shortcuts.utils

import android.net.Uri

object Validation {

    fun isAcceptableUrl(url: String) =
            !url.isBlank() &&
                    !url.equals("http://", ignoreCase = true) &&
                    !url.equals("https://", ignoreCase = true)

    fun isValidUrl(uri: Uri) =
            uri.scheme?.equals("http", ignoreCase = true) == true
                    || uri.scheme?.equals("https", ignoreCase = true) == true

}
